app/controllers/demo_mode/sessions_controller.rb

```diff
@@ -45,7 +45,7 @@ def begin_demo_if_ready!
     end
 
     def begin_demo!
-      instance_eval(&@session.begin_demo)
+      instance_eval(&@session.begin_demo.call(**query_params.to_unsafe_h.deep_symbolize_keys))
```

Comment:
This would be a breaking change, is there a better way to pass the option through?